python/ray/dataframe/dataframe.py

```diff
@@ -568,7 +568,9 @@ def first(self, offset):
         raise NotImplementedError("Not Yet implemented.")
 
     def first_valid_index(self):
-        raise NotImplementedError("Not Yet implemented.")
+        return min(
```

Comment:
I don't know if the `min` here is correct. Index doesn't need to be sorted.

Either way, we have our own `Index` object -- `self._index`. Just use the logic from Pandas to get the first valid index from that.

Reply:
resolved by getting self._index[0]